src/ray/ray_impl.rs

```diff
@@ -113,31 +113,26 @@ impl<T: BHValue, const D: usize> Ray<T, D> {
     /// where the first number is the distance from [`Ray`] to the nearest intersection point
     /// and the second number is the distance from [`Ray`] to the farthest intersection point
     ///
-    /// If there are no intersections, it returns negative one for both distances
-    pub fn intersection_slice_for_aabb(&self, aabb: &Aabb<T, D>) -> (T, T)
+    /// If there are no intersections, it returns `None`.
+    pub fn intersection_slice_for_aabb(&self, aabb: &Aabb<T, D>) -> Option<(T, T)>
```

Comment:
Hm, this is API-breaking so please document the migration path in the CHANGELOG.

Reply:
Sure! I added a changelog entry that also claimed the distance traversal was "correct." Then I realized it [might not be](https://github.com/svenstaro/bvh/issues/134). So now the wording is more precise.

Reply:
Do you wanna spend some more time on that before we merge this?